File: Monstone_Android_Client/app/src/main/java/com/example/nightwingky/monstone/main_program/me/change_user_info/ChangeUserInfoActivity.java
package com.example.nightwingky.monstone.main_program.me.change_user_info;

import android.content.Intent;
import android.os.AsyncTask;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.AdapterView;
import android.widget.Button;
import android.widget.ListView;
import android.widget.TextView;

import com.example.nightwingky.monstone.R;
import com.example.nightwingky.monstone.myConst.MyConst;

import org.json.JSONException;
import org.json.JSONObject;

import java.io.IOException;

public class ChangeUserInfoActivity extends AppCompatActivity {

    private String info_url = MyConst.getUser_change_info_url();
    private String name_url = MyConst.getUser_change_name_url();
    private String gender_url = MyConst.getUser_change_gender_url();
    private String mobile_url = MyConst.getUser_change_mobile_url();

    private ListView mListView;
    private Button btn_submit;

    public static ChangeUserInfoActivity mActivity;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_change_user_info);

        mActivity = ChangeUserInfoActivity.this;

        bindView();

        new GetUserInfoAsync().execute(info_url);
    }

    private void bindView() {
        this.mListView = (ListView) findViewById(R.id.lv_change_user_info);
        this.btn_submit = (Button) findViewById(R.id.btn_submit_change_user_info);

        mListView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                TextView tv_title;
                TextView tv_info;
                Intent intent;

                switch (position) {
                    case 1:
                        tv_title = (TextView) view.findViewById(R.id.tv_else_change_user_info);
                        tv_info = (TextView) view.findViewById(R.id.tv_info_else_change_user_info);

                        intent = new Intent(ChangeUserInfoActivity.this, ItemChangeUserInfoActivity.class);
                        intent.putExtra("title", tv_title.getText());
                        intent.putExtra("info", tv_info.getText());
                        intent.putExtra("url", name_url);
                        intent.putExtra("position", position);
                        Log.d("position", String.valueOf(position));
                        startActivity(intent);
                        break;
                    case 2:
                        tv_title = (TextView) view.findViewById(R.id.tv_else_change_user_info);
                        tv_info = (TextView) view.findViewById(R.id.tv_info_else_change_user_info);

                        intent = new Intent(ChangeUserInfoActivity.this, ItemChangeUserInfoActivity.class);
                        intent.putExtra("title", tv_title.getText());
                        intent.putExtra("info", tv_info.getText());
                        intent.putExtra("url", gender_url);
                        intent.putExtra("position", position);
                        Log.d("position", String.valueOf(position));
                        startActivity(intent);
                        break;
                    case 4:
                        tv_title = (TextView) view.findViewById(R.id.tv_else_change_user_info);
                        tv_info = (TextView) view.findViewById(R.id.tv_info_else_change_user_info);

                        intent = new Intent(ChangeUserInfoActivity.this, ItemChangeUserInfoActivity.class);
                        intent.putExtra("title", tv_title.getText());
                        intent.putExtra("info", tv_info.getText());
                        intent.putExtra("url", mobile_url);
                        intent.putExtra("position", position);
                        Log.d("position", String.valueOf(position));
                        startActivity(intent);
                        break;
                    default:
                        break;
                }
            }
        });

        btn_submit.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });
    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
        finish();
    }

    private ChangeUserInfoVO getVOByJsonData() throws IOException, JSONException {
        String jsonString = ChangeUserInfoHttp.getUserInfo(info_url, MyConst.getUser_id());

        JSONObject jsonObject = new JSONObject(jsonString);
        jsonObject = new JSONObject(jsonObject.getString("object"));

        ChangeUserInfoVO changeUserInfoVO = new ChangeUserInfoVO(
                MyConst.getUser_id(),
                jsonObject.getString("name"),
                jsonObject.getInt("gender"),
                jsonObject.getString("birth"),
                jsonObject.getString("mobile"),
                jsonObject.getString("email"),
                jsonObject.getString("portrait"),
                jsonObject.getString("introduction")
        );

        return changeUserInfoVO;
    }


    class GetUserInfoAsync extends AsyncTask<String, Void, ChangeUserInfoVO> {

        @Override
        protected ChangeUserInfoVO doInBackground(String... params) {
            try {
                Log.d("ChangeUserInfoVO", getVOByJsonData().toString());
                return getVOByJsonData();
            } catch (IOException e) {
                e.printStackTrace();
            } catch (JSONException e) {
                e.printStackTrace();
            }

            return null;
        }

        @Override
        protected void onPostExecute(ChangeUserInfoVO changeUserInfoVO) {
            super.onPostExecute(changeUserInfoVO);

            ChangeUserInfoAdapter adapter = new ChangeUserInfoAdapter(ChangeUserInfoActivity.this, changeUserInfoVO);
            mListView.setAdapter(adapter);
        }
    }

}

File: kotlin_client/app/src/main/java/com/graduation/nightwingky/book_drifting/monstone/main_program/fragmentMe/change_user_info/ChangeUserInfoActivity.kt
package com.graduation.nightwingky.book_drifting.monstone.main_program.fragmentMe.change_user_info

import android.content.Intent
import android.os.AsyncTask
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.AdapterView
import android.widget.Button
import android.widget.ListView
import android.widget.TextView

import com.graduation.nightwingky.book_drifting.R
import com.graduation.nightwingky.book_drifting.monstone.myConst.URLConst

import org.json.JSONException
import org.json.JSONObject

import java.io.IOException

class ChangeUserInfoActivity : AppCompatActivity() {

    private val info_url = URLConst.user_change_info_url
    private val name_url = URLConst.user_change_name_url
    private val gender_url = URLConst.user_change_gender_url
    private val mobile_url = URLConst.user_change_mobile_url

    private var mListView: ListView? = null
    private var btn_submit: Button? = null

    private val voByJsonData: ChangeUserInfoVO
        @Throws(IOException::class, JSONException::class)
        get() {
            val jsonString = ChangeUserInfoHttp.getUserInfo(info_url, URLConst.user_id)

            var jsonObject = JSONObject(jsonString)
            jsonObject = JSONObject(jsonObject.getString("object"))

            return ChangeUserInfoVO(
                    URLConst.user_id,
                    jsonObject.getString("name"),
                    jsonObject.getInt("gender"),
                    jsonObject.getString("birth"),
                    jsonObject.getString("mobile"),
                    jsonObject.getString("email"),
                    jsonObject.getString("portrait"),
                    jsonObject.getString("introduction")
            )
        }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_change_user_info)

        mActivity = this@ChangeUserInfoActivity

        bindView()

        GetUserInfoAsync().execute(info_url)
    }

    private fun bindView() {
        this.mListView = findViewById<View>(R.id.lv_change_user_info) as ListView
        this.btn_submit = findViewById<View>(R.id.btn_submit_change_user_info) as Button

        mListView!!.onItemClickListener = AdapterView.OnItemClickListener { parent, view, position, id ->
            val tv_title: TextView
            val tv_info: TextView
            val intent: Intent

            when (position) {
                1 -> {
                    tv_title = view.findViewById<View>(R.id.tv_else_change_user_info) as TextView
                    tv_info = view.findViewById<View>(R.id.tv_info_else_change_user_info) as TextView

                    intent = Intent(this@ChangeUserInfoActivity, ItemChangeUserInfoActivity::class.java)
                    intent.putExtra("title", tv_title.text)
                    intent.putExtra("info", tv_info.text)
                    intent.putExtra("url", name_url)
                    intent.putExtra("position", position)
                    Log.d("position", position.toString())
                    startActivity(intent)
                }
                2 -> {
                    tv_title = view.findViewById<View>(R.id.tv_else_change_user_info) as TextView
                    tv_info = view.findViewById<View>(R.id.tv_info_else_change_user_info) as TextView

                    intent = Intent(this@ChangeUserInfoActivity, ItemChangeUserInfoActivity::class.java)
                    intent.putExtra("title", tv_title.text)
                    intent.putExtra("info", tv_info.text)
                    intent.putExtra("url", gender_url)
                    intent.putExtra("position", position)
                    Log.d("position", position.toString())
                    startActivity(intent)
                }
                4 -> {
                    tv_title = view.findViewById<View>(R.id.tv_else_change_user_info) as TextView
                    tv_info = view.findViewById<View>(R.id.tv_info_else_change_user_info) as TextView

                    intent = Intent(this@ChangeUserInfoActivity, ItemChangeUserInfoActivity::class.java)
                    intent.putExtra("title", tv_title.text)
                    intent.putExtra("info", tv_info.text)
                    intent.putExtra("url", mobile_url)
                    intent.putExtra("position", position)
                    Log.d("position", position.toString())
                    startActivity(intent)
                }
                else -> {
                }
            }
        }

        btn_submit!!.setOnClickListener { finish() }
    }

    override fun onBackPressed() {
        super.onBackPressed()
        finish()
    }


    internal inner class GetUserInfoAsync : AsyncTask<String, Void, ChangeUserInfoVO>() {

        override fun doInBackground(vararg params: String): ChangeUserInfoVO? {
            try {
                Log.d("ChangeUserInfoVO", voByJsonData.toString())
                return voByJsonData
            } catch (e: IOException) {
                e.printStackTrace()
            } catch (e: JSONException) {
                e.printStackTrace()
            }

            return null
        }

        override fun onPostExecute(changeUserInfoVO: ChangeUserInfoVO) {
            super.onPostExecute(changeUserInfoVO)

            val adapter = ChangeUserInfoAdapter(this@ChangeUserInfoActivity, changeUserInfoVO)
            mListView!!.adapter = adapter
        }
    }

    companion object {

        lateinit var mActivity: ChangeUserInfoActivity
    }

}
